pkg/monitoring/prometheusagent/config.go

```diff
@@ -0,0 +1,119 @@
+package prometheusagent
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/giantswarm/observability-operator/pkg/common"
+	"github.com/giantswarm/observability-operator/pkg/monitoring"
+	"github.com/giantswarm/observability-operator/pkg/monitoring/mimir/querier"
+	"github.com/giantswarm/observability-operator/pkg/monitoring/prometheusagent/shards"
+)
+
+func (pas PrometheusAgentService) buildRemoteWriteConfig(ctx context.Context,
+	cluster *clusterv1.Cluster, logger logr.Logger, currentShards int) (*corev1.ConfigMap, error) {
+
+	organization, err := pas.OrganizationRepository.Read(ctx, cluster)
+	if err != nil {
+		logger.Error(err, "failed to get cluster organization")
+		return nil, errors.WithStack(err)
+	}
+
+	provider, err := common.GetClusterProvider(cluster)
+	if err != nil {
+		logger.Error(err, "failed to get cluster provider")
+		return nil, errors.WithStack(err)
+	}
+
+	clusterType := "workload_cluster"
+	if val, ok := cluster.Labels["cluster.x-k8s.io/cluster-name"]; ok && val == pas.ManagementCluster.Name {
+		clusterType = "management_cluster"
+	}
```

Comment:
Could be nice to have constant for "workload_cluster" and "management_cluster"

Reply:
I usually use constant if they are used more than once

Reply:
Yes sure, I bet we will need it again with next features...

Reply:
So if we need it again, I would move this in a function :D but I can do that

Reply:
Fixed as well